services/blockchain-indexer/shared/dataService/pos/validators.js

```diff
@@ -187,32 +187,46 @@ const getPosValidators = async params => {
 		meta: {},
 	};
 
-	const validatorsTable = await getValidatorsTable();
-	validators.data = await getAllValidators();
-
-	const filterBy = (list, entity) => list.filter(
-		(acc) => params[entity].includes(',')
-			? (acc[entity] && params[entity].split(',').includes(acc[entity]))
-			: (acc[entity] && acc[entity] === params[entity]),
-	);
+	const addressSet = new Set();
+	const nameSet = new Set();
 
 	if (params.publicKey) {
-		params.address = getLisk32AddressFromPublicKey(params.publicKey);
+		const address = getLisk32AddressFromPublicKey(params.publicKey);
+		params.address = params.address || address;
```

Comment:
`params.address` should not be available when the user queries with `params.publicKey`. Did we not define the param pairings in the gateway?
```suggestion
		params.address = address;
```

Reply:
we do not have param pairings defined in gateway. When called without params, this api returns a list of 10 validators. Also, param pairings validate at least one of the allowed must be passed, but do not validate that only one can be passed.

Reply:
For now, please add a TODO to verify the param pairings logic and implement the same for the endpoint. Also, set `params.address = address;` here.

Reply:
Applied.